text/0000-lifecycle-all.md

```diff
@@ -0,0 +1,109 @@
+# Meta
+[meta]: #meta
+- Name: Lifecycle All Binary
+- Start Date: 2020-01-21
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+### `/cnb/lifecycle/all`
+A new lifecycle binary with the name `lifecycle/all` will be included in each released lifecycle archive. Within a builder image, it will be found at the path `/cnb/lifecycle/all`. When invoked, it will run the following phases sequentially:
+
+* detector
+* analyzer
+* restorer
+* builder
+* exporter
+
+Each of these phases will continue to be available individually as provided by the current lifecycle.
+
+### Platform API 0.3
+The existence of `all` will constitute a non-breaking but substantive change to the Platform API, bringing us to Platform API 0.3 . Although platform will interpret this as a breaking change, we do not have a mechanism for indicating non-breaking platform API change pre 1.0. Bumping the platform API number is necessary so that platforms like `pack` know whether or no they can use this feature.
+
+# Motivation
+[motivation]: #motivation
+
+This new binary supports two main goals.
+
+### Goal 1: faster `pack` builds
+By creating a single container and invoking `/cnb/lifecycle/all` instead of creating a container per lifecycle phase, pack can shave approximate `6s` from the execution time of `pack build`.
```

Comment:
Is there a reason why this is preferable to running the command `/cnb/lifecycle/detector && /cnb/lifecycle/analyzer && /cnb/lifecycle/restorer && /cnb/lifecycle/builder && /cnb/lifecycle/exporter`?

Reply:
The benefits of `/cnb/lifecycle/all` in the `pack` vs. stringing all of the phases together are mostly aesthetic. If we pass all of the required flags to each lifecycle phase, the command starts getting very long, complicated, and redundant looking.